docs/new-architecture-playbook.md

```diff
@@ -0,0 +1,2182 @@
+
+<!-- alex ignore savage -->
+
+In general, this means you can use primitive types (strings, numbers, booleans), as well as function types, object types, and array types. Union types, on the other hand, are not supported. All types must be read-only in Flow: either `+` or `$ReadOnly<>` or `{||}` objects.
+
+> See Appendix [I. Flow Type to Native Type Mapping](#i-flow-type-to-native-type-mapping).
+
+#### Be Consistent Across Platforms and Eliminate Type Ambiguity
+
+Before adopting the new architecture in your native module, you will need to ensure your methods are consistent across platforms. This is something you will realize as you set out to write the JavaScript spec for your native module - remember, that JavaScript spec defines what the methods will look like on all supported platforms.
+
+If your existing native module has methods with the same name on multiple platforms, but with different numbers or types of arguments across platforms, you will need to find a way to make these consistent. If you have methods that can take two or more different types for the same argument, you will also need to find a way to resolve this type ambiguity as type unions are intentionally not supported.
+
+### Make sure _autolinking_ is enabled
+
+<!-- alex ignore master -->
+
+[Autolinking](https://github.com/react-native-community/cli/blob/master/docs/autolinking.md) is a feature of the React Native CLI that simplifies the installation of third-party React Native libraries. Instructions to enable _autolinking_ are available at https://github.com/react-native-community/cli/blob/master/docs/autolinking.md.
+
+#### Android
+
+On Android, this generally requires you to include `native_modules.gradle` in both your `settings.gradle[.kts]` and `build.gradle[.kts]`.
+
+If you used the default template provided with React Native (i.e. you used `yarn react-native init <Project>`), then you have autolinking already enabled.
+
+You can anyway verify that you have it enabled with:
+
+```bash
+$ grep -r "native_modules.gradle" android
+
+android/app/build.gradle:apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesAppBuildGradle(project)
+android/settings.gradle:apply from: file("../node_modules/@react-native-community/cli-platform-android/native_modules.gradle"); applyNativeModulesSettingsGradle(settings)
+...
+```
+
+#### iOS
+
+On iOS, this generally requires your library to provide a Podspec (see [`react-native-webview`](https://github.com/react-native-community/react-native-webview/blob/master/react-native-webview.podspec) for an example).
+
+> To determine if your library is set up for autolinking, check the CocoaPods output after running `pod install` on an iOS project. If you see "auto linking library name", you are all set to go.
+
+### Preparing your JavaScript codebase for the new React Native Renderer (Fabric)
```

Comment:
By the time I was in this section, I forgot we're still discussing library code. I read "your JavaScript codebase" as a cue that we're talking about the app now. Can we make this something like "Preparing your library code for ..." so that this context is not lost? It's hard to track with multiple levels of header nesting.

Reply:
You're totally right on this point. We're going to move this part out of the "Library" section and just move it to its own page. This will be done in a subsequent PR.